src/python/pants/backend/experimental/python/user_lockfiles.py

```diff
@@ -46,15 +46,18 @@ async def generate_user_lockfile_goal(
         return GenerateUserLockfileGoal(exit_code=1)
 
     transitive_targets = await Get(TransitiveTargets, TransitiveTargetsRequest(addresses))
-    reqs = PexRequirements.create_from_requirement_fields(
-        tgt[PythonRequirementsField]
-        # NB: By looking at the dependencies, rather than the closure, we only generate for
-        # requirements that are actually used in the project.
-        for tgt in transitive_targets.dependencies
-        if tgt.has_field(PythonRequirementsField)
-    )
+    req_strings = PexRequirements.create_from_requirement_fields(
+        (
+            tgt[PythonRequirementsField]
+            # NB: By looking at the dependencies, rather than the closure, we only generate for
+            # requirements that are actually used in the project.
+            for tgt in transitive_targets.dependencies
+            if tgt.has_field(PythonRequirementsField)
+        ),
+        constraints_strings=(),
```

Comment:
This is the default - why set it? Ditto a couple other places.

Reply:
I made `constraints_strings` a non-optional argument for `PexRequirements.create_from_requirement_fields` in this change, because of the kind of question above with `mypy`: when requirements are being created from `targets`, there should always be an explicit question of whether to use the constraints.